src/pytorch_lightning/plugins/precision/apex_amp.py

```diff
@@ -97,7 +87,7 @@ def optimizer_step(
         self._after_closure(model, optimizer, optimizer_idx)
         skipped_backward = closure_result is None
         # in manual optimization, the closure does not return a value
-        if not isinstance(model, pl.LightningModule) or not model.automatic_optimization or not skipped_backward:
```

Comment:
Removed several isinstance checks in the plugins. These were here to separate code path for Lite and PL, but we no longer have them shared after #14735. The code path is now completely isolated.